### PR TITLE
MYST-498 Cancel connection returns 499

### PR DESCRIPTION
**Pull request by donce**

### CODE REVIEW

File: client/connection/manager.go [hunk 1]
```diff
@@ -62,9 +62,17 @@ func (manager *connectionManager) Connect(consumerID, providerID identity.Identi
 		}
 	}()
 
+	err = manager.startConnection(consumerID, providerID)
+	if err == utils.ErrRequestCancelled {
+		return ErrConnectionCancelled
```
Comment (tadovas) [inline client/connection/manager.go]: Do we really need separate error in manager if utils.ErrRequestCancelled is the only error which indicates cancelation ?

Comment (donce) [reply client/connection/manager.go]: Yes - we specify more concrete custom error message, which we use in our api response body.

Comment (donce) [reply client/connection/manager.go]: I was wondering, maybe this mapping can be done in api level - if manager.Connect returns `ErrRequestCancelled`, than we return custom error with response body we want.

Comment (donce) [reply client/connection/manager.go]: @tadovas pushed new commit of what I've meant.

File: client/connection/manager.go [hunk 2]
```diff
@@ -20,8 +20,6 @@ var (
 	ErrNoConnection = errors.New("no connection exists")
 	// ErrAlreadyExists error indicates that aciton applieto to manager expects no active connection (i.e. connect)
 	ErrAlreadyExists = errors.New("connection already exists")
-	// ErrConnectionCancelled indicates that connection in progress was cancelled by request of api user
-	ErrConnectionCancelled = errors.New("connection was cancelled")
```
Comment (Waldz) [inline client/connection/manager.go]: What was wrong with more concrete error?

File: client/connection/manager.go [hunk 3]
```diff
@@ -214,7 +216,7 @@ func (manager *connectionManager) waitForConnectedState(stateChannel <-chan open
 				manager.onStateChanged(state, sessionID)
 			}
 		case <-cancelRequest:
-			return ErrConnectionCancelled
+			return utils.ErrRequestCancelled
```
Comment (Waldz) [inline client/connection/manager.go]: Dont like that internally used library (`cancellable`) is exposed to external world

**Pull request merged**